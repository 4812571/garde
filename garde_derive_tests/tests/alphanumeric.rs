#[path = "./util/mod.rs"]
mod util;

#[derive(Debug, garde::Validate)]
struct Test<'a> {
    #[garde(alphanumeric)]
    field: &'a str,
}

#[test]
fn alphanumeric_valid() {
    util::check_ok(&[Test { field: "abcd0123" }], &())
}

#[test]
fn alphanumeric_invalid() {
    util::check_fail!(&[Test { field: "!!!!" }], &())
}
