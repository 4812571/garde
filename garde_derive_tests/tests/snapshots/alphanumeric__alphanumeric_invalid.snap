---
source: garde_derive_tests/tests/alphanumeric.rs
expression: snapshot
---
Test {
    field: "!!!!",
}
`field` is not alphanumeric


